### PR TITLE
Post-IODC Updates

### PR DESCRIPTION
**Pull request by timgdavies**
Following IODC I agreed to stage some updates to the working draft / alpha before work starts on initial pilot implementation work, and developing implementation guidance,

I've included those changes here, and will update the current gh-pages branch to reflect this PR. 

I've not covered absolutely everything discussed, but hopefully this moves things forward enough to support next steps during late October / November.

### CODE REVIEW

File: schema/project-level/codelists/votingRights.csv [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Code,Title,Description
```
Comment (jpmckinney) [inline schema/project-level/codelists/votingRights.csv]: Why are the titles and descriptions missing from this codelist, whose codes match those in the OCDS shareholders extension?

File: schema/project-level/codelists/+documentType.csv [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Code,Title,Description
```
Comment (jpmckinney) [inline schema/project-level/codelists/+documentType.csv]: There is both `+documentType.csv` and `documentType.csv` in the same folder. Shouldn't only one be retained?

File: mapping/process-level.csv [hunk 3]
```diff
@@ -0,0 +1,20 @@
+CoST IDS Element,Draft definition,Maps to,Discussion
+Procuring entity,Enter name of the organisation carrying out the procurement,Project Level: Add an entry to ``parties`` with 'procuringEntity' included in its ``.roles`` | Contracting process: Record the name and identifier in ``contractingProcesses.procuringEntity``,
+Procuring entity contact details,Postal and Electronic address,Project Level: Add an entry to ``parties`` with 'procuringEntity' included in its ``.roles`` | Contracting process: record the name and identifier in `contractingProcesses.procuringEntity`,
+Contract administrative entity,Enter name of the organisation carrying out the contract administrative entity if different from the Procuring Entity,Project Level: Add an entry to ``parties`` with 'administativeEntity' included in its ``.roles`` | Contracting process: Record the name and identifier in `contractingProcesses.administrativeEntity`,
```
Comment (jpmckinney) [inline mapping/process-level.csv]: Typo: Need double-backticks, as I assume this is interpreted as RST and not Markdown.

File: schema/project-level/codelists/+documentType.csv [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+projectScope,Project scope,"A description of the main outputs from the project that are being taken forward into construction (including type, quantity and units)"
+landAndSettlementImpact,Land and Settlement Impact,"State the amount of land and property that was acquired for the project (e.g. 25sq km land), and outline related impacts (e.g. archaeological issues (moved saxon burial site), local/indigenous settlements (relocated 5 indigenous villages of 500 villagers each), impacts on local businesses e.g. (30 business properties purchased))."
+environmentalImpact,Environmental Impact,"Describe the environmental impacts and
```
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply schema/project-level/codelists/+documentType.csv]: The description used here is taken from CoST Guidance Note 9

File: schema/project-level/codelists/projectPhase.csv [hunk 5]
```diff
@@ -0,0 +1,7 @@
+Code,Title,Description
```
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply schema/project-level/codelists/projectPhase.csv]: Addressed in https://github.com/open-contracting/infrastructure/commit/74497461b33d878183c5b6e9aa3c5a09edeadd7c

File: docs/projects/reference.md [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+```eval_rst
+
+   .. csv-table::
+      :header-rows: 1
+      :class: codelist-table
+      :file: ../../build/current_lang/codelists/method.csv
+
+```
+
+### ProjectPhase
```
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply docs/projects/reference.md]: Addressed in https://github.com/open-contracting/infrastructure/commit/708fe7a2c953b9d1d9a68321ceb00ef6821307b6

File: schema/project-level/codelists/+documentType.csv (hunk 2 above)
Comment (duncandewhurst) [reply schema/project-level/codelists/+documentType.csv]: See https://github.com/open-contracting/infrastructure/issues/25

File: mapping/process-level.csv [hunk 7]
```diff
@@ -0,0 +1,20 @@
+CoST IDS Element,Draft definition,Maps to,Discussion
+Procuring entity,Enter name of the organisation carrying out the procurement,Project Level: Add an entry to ``parties`` with 'procuringEntity' included in its ``.roles`` | Contracting process: Record the name and identifier in ``contractingProcesses.procuringEntity``,
+Procuring entity contact details,Postal and Electronic address,Project Level: Add an entry to ``parties`` with 'procuringEntity' included in its ``.roles`` | Contracting process: record the name and identifier in `contractingProcesses.procuringEntity`,
```
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply mapping/process-level.csv]: Addressed in https://github.com/open-contracting/infrastructure/commit/66fa0b19a3c440bb1f2b175c88a0c1666f062486

File: mapping/process-level.csv [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+Procuring entity contact details,Postal and Electronic address,Project Level: Add an entry to ``parties`` with 'procuringEntity' included in its ``.roles`` | Contracting process: record the name and identifier in `contractingProcesses.procuringEntity`,
+Contract administrative entity,Enter name of the organisation carrying out the contract administrative entity if different from the Procuring Entity,Project Level: Add an entry to ``parties`` with 'administativeEntity' included in its ``.roles`` | Contracting process: Record the name and identifier in `contractingProcesses.administrativeEntity`,
+Contract status,"Select from pre-award, active or closed",Contracting process: Publish as ``contractingProcesses.status`` using the processStatus codelist. ,`Issue #1 <https://github.com/open-contracting/infrastructure/issues/1>`_
+Procurement process,"Develop a list such as International Competitive Bidding, National Competitive Bidding, Donor Procurement Rules, Framework, Direct Award",Contracting process: Publish as ``contractingProcess.procurementMethodDetails`` and map to ``contractingProcess.procurementMethod``. ,
```
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply mapping/process-level.csv]: Addressed in https://github.com/open-contracting/infrastructure/commit/66fa0b19a3c440bb1f2b175c88a0c1666f062486

File: mapping/process-level.csv [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+Contract status,"Select from pre-award, active or closed",Contracting process: Publish as ``contractingProcesses.status`` using the processStatus codelist. ,`Issue #1 <https://github.com/open-contracting/infrastructure/issues/1>`_
+Procurement process,"Develop a list such as International Competitive Bidding, National Competitive Bidding, Donor Procurement Rules, Framework, Direct Award",Contracting process: Publish as ``contractingProcess.procurementMethodDetails`` and map to ``contractingProcess.procurementMethod``. ,
+Contract type,"Develop a list such as Design, Supervision, Design & Supervision, Design & Build, Construction","Contract process: Add one or more values to the ``contractingProcess.focus`` array. (e.g. ['design','build'] for a design and build contract).",`Issue #11 <https://github.com/open-contracting/infrastructure/issues/11>`_
```
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply mapping/process-level.csv]: Addressed in https://github.com/open-contracting/infrastructure/commit/66fa0b19a3c440bb1f2b175c88a0c1666f062486

File: mapping/process-level.csv (hunk 3 above)
Comment (duncandewhurst) [reply mapping/process-level.csv]: Addressed in https://github.com/open-contracting/infrastructure/commit/66fa0b19a3c440bb1f2b175c88a0c1666f062486

File: mapping/project-level.csv [hunk 10]
```diff
@@ -0,0 +1,22 @@
+CoST IDS Element,Draft definition,Maps to,Discussion
+Project owner,Name of the sponsoring Government department,Project Level: Add an entry to ``parties`` with 'owner' included in its ``.roles``.,
+Sector,"Develop a list of sectors relevant to country e.g. housing, transport, energy, water etc.","Project Level: Publish as ``sector``, with any additional sectors in ``additionalClassifications``.",`Issue #16 <https://github.com/open-contracting/infrastructure/issues/16>`_
+Subsector,"Develop a subset for each sector e.g. Transport could be subdivided into national highway, local road, railway, port, airport etc.",Publish in ``additionalClassifications``. ,`Issue #16 <https://github.com/open-contracting/infrastructure/issues/16>`_
```
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply mapping/project-level.csv]: Addressed in https://github.com/open-contracting/infrastructure/commit/db3987d23e714a002aed4ee71946131e3469aff6

File: mapping/project-level.csv [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+Funding sources,Name the funding organisation(s)/sources of funding,"Project Level: Include each of the funding organisations in the ``parties`` array with 'funder' in ``.roles``. When information on the amount provided by each funder is available, use ``budget/budgetBreakdown`` to provide details.",
+Project Budget,"Specify the projected costs/allocated budget for the project (currency and amount). The budget includes land / property acquisition, environmental mitigation measures, H&S provisions, client, consultant & contractor costs, VAT etc.",Project Level: Publish as ``budget/amount``.,
+Project budget approval date,Date project budget was authorised,Projet Level: Publish as ``budget/approvalDate`` and include additional document with ``.type`` of 'budgetApproval' if required,
+Project status (current),"The current stage of the project. Select from identification, preparation, construction, completion, completed or cancelled.",Project Level: Publish as ``status``.,`Issue #10 <https://github.com/open-contracting/infrastructure/issues/10>`_
+Project completion) cost,"State projected or actual completion cost
```
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply mapping/project-level.csv]: Addressed in https://github.com/open-contracting/infrastructure/commit/db3987d23e714a002aed4ee71946131e3469aff6

File: mapping/project-level.csv [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+Contact details,Postal and electronic address of the Project Owner,Project Level: Include ``.address`` and ``.contactPoint`` information for the Party with the 'owner' in ``.roles``. ,
+Funding sources,Name the funding organisation(s)/sources of funding,"Project Level: Include each of the funding organisations in the ``parties`` array with 'funder' in ``.roles``. When information on the amount provided by each funder is available, use ``budget/budgetBreakdown`` to provide details.",
+Project Budget,"Specify the projected costs/allocated budget for the project (currency and amount). The budget includes land / property acquisition, environmental mitigation measures, H&S provisions, client, consultant & contractor costs, VAT etc.",Project Level: Publish as ``budget/amount``.,
+Project budget approval date,Date project budget was authorised,Projet Level: Publish as ``budget/approvalDate`` and include additional document with ``.type`` of 'budgetApproval' if required,
```
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply mapping/project-level.csv]: Addressed in https://github.com/open-contracting/infrastructure/commit/db3987d23e714a002aed4ee71946131e3469aff6

File: mapping/project-level.csv (hunk 11 above)
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply mapping/project-level.csv]: Addressed in https://github.com/open-contracting/infrastructure/commit/db3987d23e714a002aed4ee71946131e3469aff6

File: mapping/project-level.csv [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+Project status (current),"The current stage of the project. Select from identification, preparation, construction, completion, completed or cancelled.",Project Level: Publish as ``status``.,`Issue #10 <https://github.com/open-contracting/infrastructure/issues/10>`_
+Project completion) cost,"State projected or actual completion cost
+(currency and amount)",Project Level: Publish as ``completion/finalValue``,
+Completion date,State projected or actual completion date,Project Level: Publish as ``completion/endDate``,
+Project Scope at completion (projected),"Indicate projected or actual scope of project. Aim is to show if the completed project scope differs from the original project scope. Specify main outputs (type, quantity, unit)",ProjectLevel: Publish free text as ``completion/finalScopeDetails`` and/or include document with ``.type`` of ``scope`` and dates and descriptions that show this is final scope at completion.,
```
[the remark this answers is not in the document]
Comment (duncandewhurst) [reply mapping/project-level.csv]: Addressed in https://github.com/open-contracting/infrastructure/commit/db3987d23e714a002aed4ee71946131e3469aff6

**Comment by duncandewhurst** [comment]
I've addressed the comments from @jpmckinney and merged this branch into `master` so that future development work can take place in branches from there.

I'll update the `gh-pages` branch with these changes now too

File: mapping/project-level.csv (hunk 11 above)
[the remark this answers is not in the document]
Comment (jpmckinney) [reply mapping/project-level.csv]: Hmm, I still see "(currency and amount)" on a new line, which should instead go at the end of the previous line.

[the remark this answers is not in the document]
Comment (jpmckinney) [reply mapping/project-level.csv]: I still see "Project completion) cost" in that commit, instead of "Project completion cost".

**Comment by jpmckinney** [comment]
In case my commit comments are hard to track, I also noted:

* I still see "(currency and amount)" on a new line, which should instead go at the end of the previous line.
* I still see "Project completion) cost" in that commit, instead of "Project completion cost".

**Pull request merged**